src/hugo/content/en-us/patterns/MVC/_index.md

```diff
@@ -0,0 +1,88 @@
+
+## Considerations
+
+### Safety
+
+In many systems, particularly those that actuate physical systems like motors, solenoids, or relays, pathways for locking out or failing safe for those systems should be kept as short as possible. When at all possible, these lockout systems and sensors should be configured so that they disable actuation of potentially dangerous systems. For example, place a limit switch in series with a motor activation circuit. This ensures that no software failure, attack, or disintermediation can defeat the designed safety controls. 
+
+Because of this critical relationship of the safety controls to device operation, the status of the sensors and. controls should be monitored and reported. Here too, care must be taken to ensure that the monitoring of the signal cannot in any way affect the function of the control. One common way to do this would be to use a buffer, follower, or open drain device to isolate the control signal from the measurement system. Consider if the signals were to be routed to GPIO pins, such a buffer would prevent a misconfiguration of the pinmux to output from defeating the safety control. 
+
+For any device and system, first review the components and subsystems as to how they affect the overall safety of the system and minimize the potential impact of software errors or attacks to the system. A similar discussion of security and attack surfaces should also be undertaken, but that is beyond the scope of this document.
+
+### Local Buttons
+
+When the user pushes the local button (on a GPIO input for example), how shall this be processed? We desire to have low-latency response and the sytem should function without cloud connection. But when we do have connectivity, the mobile app should show the status also with low latency. 
+
+The button state (up/down/edge/etc) should be monitored and maintained by the **model**. Subscribed **view**s can monitor these signals. The first **view** would request a 'local' **protocol** from the **connection** and report state, while the second **view** would report to any remote systems. The 'local' **protocol** implementation would update the **connection** managed _Observable_ to which a **controller** is subscribed and would take appropriate action.  The full separation of concerns is maintained, while full local-only functionality is maintained and any remote systems kept up to date.
```

Comment:
Would having the steps as ordered or unordered list be easier to read?

Reply:
heh heh... i had it that way and flattened....